src/collection.js

```diff
@@ -515,19 +515,17 @@ export default class Collection {
    * @return {Promise}
    */
   _updateRaw(oldRecord, newRecord, {synced = false} = {}) {
-    let updated;
     // Make sure to never loose the existing timestamp.
     if (oldRecord && oldRecord.last_modified && !newRecord.last_modified) {
       newRecord.last_modified = oldRecord.last_modified;
     }
     // If only local fields have changed, then keep record as synced.
     const isIdentical = oldRecord && recordsEqual(oldRecord, newRecord, this.localFields);
     const keepSynced = isIdentical && oldRecord._status == "synced";
-    let newStatus = (keepSynced || synced) ? "synced" : "updated";
-    if (!oldRecord) {
-      newStatus = "created";
-    }
-    updated = markStatus(newRecord, newStatus);
+    const neverSynced = oldRecord && oldRecord._status == "created";
+    const newStatus = (keepSynced || synced) ? "synced"
+                                             : neverSynced ? "created" : "updated";
+    const updated = markStatus(newRecord, newStatus);
```

Comment:
I personally find this logic kind of ugly and I think we could actually simplify `_updateRaw` a lot more than this. Rather than adding all this code to try to preserve the `oldRecord`'s status, let's just identify the cases where we want to *change* the `newRecord`'s status. This is OK so long as the user doesn't do anything crazy with the status, like update a record and set it to `updated` or something. (If they do, the existing code will try to cover it up; my proposal would be to let it go and get mysterious errors later.)

First, `newRecord` should always have a status -- if it doesn't, we should copy the old status; if there was no old status, newRecord should be "created" and we're done.

As far as I can tell, the only cases where we need to change `newRecord`'s status are:

1. when the old record's status was synced but the new record has changed, `newRecord` should be set to "updated"
2. when we get a `synced` option, in which case we should force the status of the new record to be "synced"

I think this will be a lot cleaner than nested ternary operators.